homework/src/index.js

```diff
@@ -31,14 +30,92 @@
+    const root = document.getElementById('root');
+    const div = createAndAppend('div', root, { class: 'data' });
+    const header = createAndAppend('p', div, { text: 'HYF Repositories', class: 'header' });
+    const table = createAndAppend('table', root, { class: 'table' });
+    const select = createAndAppend('select', header, { id: 'list' });
+    const tbody = createAndAppend('tbody', table);
+    const tr = createAndAppend('tr', tbody, { text: 'Repository: ', class: 'label' });
+    const repositoryLink = createAndAppend('a', tr);
+    const description = createAndAppend('tr', tbody, { class: 'label' });
+    const forks = createAndAppend('tr', tbody, { class: 'label' });
+    const container = createAndAppend('div', root, { id: 'container' });
+    const updated = createAndAppend('tr', tbody, { class: 'label' });
+
+    fetchJSON(url, (error, repositories) => {
+      if (error) {
+        createAndAppend('div', container, {
+          text: error.message,
+          class: 'alert-error'
+        });
       } else {
-        createAndAppend('pre', root, { text: JSON.stringify(data, null, 2) });
+        repositories.sort((a, b) => a.name.localeCompare(b.name));
+        repositories.forEach((item, index) => {
+          createAndAppend('option', select, { 'text': item.name, 'value': index });
+
+          renderRepositoryBox(
+            repositories[0],
+            repositoryLink,
+            description,
+            forks,
+            updated
+          );
+        });
+      }
+
+      select.addEventListener('change', () => {
+        container.innerHTML = "";
+        const index = event.target.value;
+
+        renderRepositoryBox(
+          repositories[select.value],
+          repositoryLink,
+          description,
+          forks,
+          updated
+        );
+        const contributorOnSelect = repositories[index].contributors_url;
+        fetchJSON(contributorOnSelect, (err, contributorData) => {
+          renderContributors(contributorData, container);
+        });
+      });
+      const contributorsInfo = repositories[0].contributors_url;
+      fetchJSON(contributorsInfo, (error, contributorData) => {
+        renderContributors(contributorData, container);
+      });
+
+      function renderRepositoryBox(
+        repositories,
+        repositoryLink,
+        forks,
+        description,
+        updated,
+      ) {
+        repositoryLink.innerText = repositories.name;
+        repositoryLink.setAttribute('href', repositories.html_url);
+        repositoryLink.setAttribute('target', '_blank');
+        forks.innerHTML = 'Forks: ' + repositories.forks;
+        description.innerHTML = 'Description: ' + repositories.description;
+        updated.innerHTML = 'Updated: ' + new Date(repositories.updated_at).toLocaleString();
       }
+
     });
+    function renderContributors(data, container) {
+      const div = createAndAppend('div', container, { class: 'right_div' });
```

Comment:
The class `right_div` does not have any styles and it does not appear on the right side...